### PR TITLE
Improve error handling on .pem root CA cert import

### LINKED ISSUE #2526: CA Certs errors
HI, I am getting errors each time i try to import my root CA certs into zap 2.4.3. This is required in order to test in a mutual ssl communication. the stack trace can be seen below...

`java.lang.ArrayIndexOutOfBoundsException: 1
    at org.zaproxy.zap.extension.dynssl.SslCertificateUtils.parseDERFromPEM(Unknown Source)
    at org.zaproxy.zap.extension.dynssl.SslCertificateUtils.pem2Keystore(Unknown Source)
    at org.zaproxy.zap.extension.dynssl.DynamicSSLPanel.doImport(Unknown Source)
    at org.zaproxy.zap.extension.dynssl.DynamicSSLPanel.access$500(Unknown Source)
    at org.zaproxy.zap.extension.dynssl.DynamicSSLPanel$5.actionPerformed(Unknown Source)
    at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
    at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2346)
    at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
    at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
    at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
    at java.awt.Component.processMouseEvent(Component.java:6525)
    at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
    at java.awt.Component.processEvent(Component.java:6290)
    at java.awt.Container.processEvent(Container.java:2234)
    at java.awt.Component.dispatchEventImpl(Component.java:4881)
    at java.awt.Container.dispatchEventImpl(Container.java:2292)
    at java.awt.Component.dispatchEvent(Component.java:4703)
    at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4898)
    at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4533)
    at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4462)
    at java.awt.Container.dispatchEventImpl(Container.java:2278)
    at java.awt.Window.dispatchEventImpl(Window.java:2750)
    at java.awt.Component.dispatchEvent(Component.java:4703)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
    at java.awt.EventQueue.access$500(EventQueue.java:97)
    at java.awt.EventQueue$3.run(EventQueue.java:709)
    at java.awt.EventQueue$3.run(EventQueue.java:703)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)
    at java.awt.EventQueue$4.run(EventQueue.java:731)
    at java.awt.EventQueue$4.run(EventQueue.java:729)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
    at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:184)
    at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:229)
    at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:227)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:227)
    at java.awt.Dialog.show(Dialog.java:1084)
    at java.awt.Component.show(Component.java:1654)
    at java.awt.Component.setVisible(Component.java:1606)
    at java.awt.Window.setVisible(Window.java:1014)
    at java.awt.Dialog.setVisible(Dialog.java:1005)
    at org.parosproxy.paros.extension.AbstractDialog.setVisible(Unknown Source)
    at org.parosproxy.paros.view.AbstractParamDialog.showDialog(Unknown Source)
    at org.parosproxy.paros.control.MenuToolsControl.options(Unknown Source)
    at org.parosproxy.paros.control.MenuToolsControl.options(Unknown Source)
    at org.parosproxy.paros.view.MainMenuBar$1.actionPerformed(Unknown Source)
    at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
    at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2346)
    at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
    at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
    at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
    at com.apple.laf.ScreenMenuItem.actionPerformed(ScreenMenuItem.java:125)
    at java.awt.MenuItem.processActionEvent(MenuItem.java:669)
    at java.awt.MenuItem.processEvent(MenuItem.java:628)
    at java.awt.MenuComponent.dispatchEventImpl(MenuComponent.java:351)
    at java.awt.MenuComponent.dispatchEvent(MenuComponent.java:339)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:761)
    at java.awt.EventQueue.access$500(EventQueue.java:97)
    at java.awt.EventQueue$3.run(EventQueue.java:709)
    at java.awt.EventQueue$3.run(EventQueue.java:703)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)
    at java.awt.EventQueue$4.run(EventQueue.java:731)
    at java.awt.EventQueue$4.run(EventQueue.java:729)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)`

Is there a way to do this from source?

### PR DESCRIPTION
**Pull request by thc202**
Change DynamicSSLPanel to warn/inform if the .pem file does not contain
the required sections (certificate and private key), also, warn if the
sections are not properly base64 encoded.
Change SslCertificateUtils to support the previous changes.
Add some tests to assert the expected behaviour of SslCertificateUtils.

Fix #2526 - CA Certs errors

### CODE REVIEW

**Comment by thc202** [comment]
Didn't notice your commit. Will add that now.

**Comment by thc202** [comment]
Change done.

File: test/org/zaproxy/zap/extension/dynssl/SslCertificateUtilsUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+import org.junit.Test;
+
+/**
+ * Unit test for {@link SslCertificateUtils}.
+ */
+public class SslCertificateUtilsUnitTest {
+
+    private static final String CERT_DATA = "Certificate data...";
+    private static final String CERT_DATA_BASE64 = "Q2VydGlmaWNhdGUgZGF0YS4uLg==";
```
Comment (kingthorin) [inline test/org/zaproxy/zap/extension/dynssl/SslCertificateUtilsUnitTest.java]: Would it be better to b64 encode during execution? That way if the value of `CERT_DATA` is change then the encoded version doesn't have to be updated? (Same for the key below.)

Comment (thc202) [inline test/org/zaproxy/zap/extension/dynssl/SslCertificateUtilsUnitTest.java]: Agreed.

**Comment by thc202** [comment]
Comment addressed. Thank you!

**Pull request merged**